### PR TITLE
[NEW UI] Start TS

### PR DESCRIPTION
**Pull request by ga-devfront**
| Questions         | Answers
| ------------------| -------------------------------------------------------
| Description?      | Initialize the TS for hydration pages (home page is fully fonctionnal now !)
| Type?             | new feature
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | N/A
| Sponsor company   | @PrestaShopCorp
| How to test?      | You need to build the TS (see readme) and try to access to home page and navigate to update page. You can try next en before from your browser too.

### CODE REVIEW

File: _dev/src/ts/pages/Home.ts [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+  private handleSubmit = (event: Event) => {
+    event.preventDefault();
+    if (this.form) {
+      const formData = new FormData(this.form);
+
+      const route = this.form.dataset.route;
+
+      if (route) {
+        new RequestHandler().post(route, formData);
```
Comment (M0rgan01) [inline _dev/src/ts/pages/Home.ts]: We could put the request handler in a static variable or in the global context, so as not to have to re-instantiate it each time

File: _dev/src/ts/api/RequestHandler.ts [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+import { ApiResponse } from '../types/apiTypes';
+import Hydration from '../utils/Hydration';
+
+export default class RequestHandler {
+  constructor() {
+    window.AutoUpgrade.classes.RequestHandler = this;
+  }
+
+  public post(route: string, data = new FormData(), fromPopState?: boolean) {
+    if (data) {
```
Comment (Quetzacoalt91) [inline _dev/src/ts/api/RequestHandler.ts]: I guess this test is always true, no need to check it.

File: _dev/src/ts/pages/Home.ts [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+  submitButton: HTMLButtonElement | null;
+
+  constructor() {
+    super();
+    this.form = document.forms.namedItem('next_page');
+    this.submitButton = this.getSubmitButton();
+  }
+
+  public mount = () => {
+    if (this.form) {
```
Comment (Quetzacoalt91) [inline _dev/src/ts/pages/Home.ts]: If the form is not found, can we consider this as normal? Or shall we trigger an exception?

File: _dev/src/ts/pages/Home.ts [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+      this.submitButton?.removeAttribute('disabled');
+    } else {
+      this.submitButton?.setAttribute('disabled', 'true');
+    }
+  };
+
+  private handleSubmit = (event: Event) => {
+    event.preventDefault();
+    if (this.form) {
+      const formData = new FormData(this.form);
```
Comment (Quetzacoalt91) [inline _dev/src/ts/pages/Home.ts]: Move the initialization inside the if, otherwise you may define it for nothing.

File: _dev/src/ts/pages/Home.ts [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+
+      const route = this.form.dataset.route;
+
+      if (route && window.AutoUpgrade.classes.RequestHandler) {
+        window.AutoUpgrade.classes.RequestHandler.post(route, formData);
```
Comment (Quetzacoalt91) [inline _dev/src/ts/pages/Home.ts]: If you're running a file of the autoupgrade project, is this relevant to always check everything is defined?

And if we regularly call this method, what about having an helper to reach it faster?
Something like `Api.post(...);`?

File: _dev/src/ts/routing/RouteHandler.ts [hunk 6]
```diff
@@ -0,0 +1,42 @@
+export default class RouteHandler {
+  constructor() {
+    window.AutoUpgrade.classes.RouteHandler = this;
```
Comment (Quetzacoalt91) [inline _dev/src/ts/routing/RouteHandler.ts]: Is this a good practice? Shouldn't it be better for the caller of this class to be responsible in defining it in the global domain?

File: ajax-upgradetab.php [hunk 7]
```diff
@@ -37,6 +37,8 @@
  * Calling it from the module/autoupgrade folder will have unwanted consequences on the upgrade and your shop.
  */
 require_once realpath(dirname(__FILE__) . '/../../modules/autoupgrade') . '/ajax-upgradetabconfig.php';
+// TODO: We need to remove this in the futur and add Twig into the module dependencies.
```
Comment (Quetzacoalt91) [inline ajax-upgradetab.php]:
```suggestion
// TODO: We need to remove this in the future and add Twig into the module dependencies.
```

**Review by Quetzacoalt91 — DISMISSED** [review]

**Review by M0rgan01 — DISMISSED** [review]

**Review by tblivet — APPROVED** [review]
👌
